packages/workspace/src/browser/workspace-frontend-contribution.ts

```diff
@@ -97,6 +104,40 @@ export class WorkspaceFrontendContribution implements CommandContribution, Keybi
             new UriAwareCommandHandler(this.selectionService, {
                 execute: (uri: URI) => this.saveAs(uri),
             }));
+        commands.registerCommand(WorkspaceCommands.UPLOAD, new FileSelection.CommandHandler(this.selectionService, {
+            multi: false,
+            isEnabled: selection => this.isEnabled(selection),
+            isVisible: selection => this.isVisible(selection),
+            execute: selection => this.upload(selection)
+        }));
+    }
+
+    protected isEnabled(selection: FileSelection | undefined): boolean {
```

Comment:
please inline `isEnabled` and `isVisible`, such method names don't make sense in `WorkspaceFrontendContribution` or rename it to `isUploadEnabled`and `isUploadVisible`